### DISCUSSION TITLE
Microsoft plans to stop 3rd party apps like ExplorerPatcher and more

### DISCUSSION THREAD

**Discussion by FarmingCrops** [valinet/ExplorerPatcher#2538] [General]
as of the new windows 11 insider build ExplorerPatcher does not work as microsoft is trying to block Explorer to start when ExplorerPatcher is installed.

https://www.youtube.com/watch?v=Uwd20bmJvT0

**Comment by zett1201** [comment]
they always try to, but fail
i remember when they made the ribbon customization unable to work on the latest 22h2 months ago.
23h2 works fine (for now) so i stick to 21h2

**Comment by pyrates999** [comment]
Canary builds are not supported because the code is not final and can change with each new release.

**Comment by johnwayne4d** [comment]
Now there is this: [https://winaero.com/microsoft-is-finally-killing-the-windows-10-taskbar-code-in-windows-11/amp/](https://winaero.com/microsoft-is-finally-killing-the-windows-10-taskbar-code-in-windows-11/amp/) which unless Devs is working on a complete rewrite, would do patcher in for good. I came here hoping to find an update from Devs.

**Reply by pyrates999** [reply]
It prevents the code from loading, but something has to be there to prevent the code from loading.  So there's still a possibility of enabling it.

**Reply by johnwayne4d** [reply]
Oh good, so there is hope, thanks for that info.

**Reply by pyrates999** [reply]
I wouldn't call it hope.  More like preventing the code from loading this would be something I would do before removing the code entirely.

**Reply by FarmingCrops** [reply]
24h2 now stops ExplorerPatcher

https://www.youtube.com/watch?v=g0WGO_1BjwM&t

**Comment by christopbader** [comment]
I use EP on all 5 of my PCs all on 23h2, let's hope there's a fix to MS's interference to third-party programs as I rely on EP for customization

**Comment by pyrates999** [comment]
duplicate of: https://github.com/valinet/ExplorerPatcher/issues/3085

please close this.

https://fxtwitter.com/StartIsBack/status/1775660727719928173

You're not supposed to use ExplorerPatcher on Dev/Canary at the moment.

**Comment by wardhale** [comment]
I've been testing 24H2 on a VM and I know that there's no support for dev/canary builds... but explorer patcher does work to an extent. You just have to rename the installation exe file to install it. You can enable Windows 10/7 file explorers, change most settings, but a few things don't work... most importantly for me, the taskbar can not be resized (no small icons options). Registry hacks no longer work either... However, Start11 is able to take over the taskbar and resize it. The system tray is still the Windows 11 nightmare, but at least they are able to resize the taskbar. So, that gives me hope... it can be done. We just can't rely on using the old Windows 10 taskbar/explorer.

**Reply by christopbader** [reply]
Now on EP's latest version 65.2 and seems good on 23h2,  on all 5 machines, the main thing for me is the File Explorer Ribbon and Start menu also the context menus, I understand the issues with the taskbar but I can live with what ever T-Bar is available

**Comment by StavrosTsik** [comment]
I don't know if the developer has the ability/willing/somehow, to brink back Win10 TaskBar for future versions of Windows 11. :worried::worried:

**Comment by wardhale** [comment]
> I don't know if the developer has the ability/willing/somehow, to brink back Win10 TaskBar for future versions of Windows 11. 😟😟

I'm not sure if it's possible to bring back the Win10 TaskBar because it's been removed from the OS. I restored it by copying explorer.exe from my Windows 10 machine to the Windows 11 VM, then deleting the Windows 11 explorer.exe and replacing it. It actually worked, but the system tray was missing the networking icon, and the language options always pop up in the top left of the screen. 

Even if that could be fixed, you can't take things from Win10 and package it into ExplorerPatcher... MS would probably sue. So, the only real way forward is to break reliance on legacy elements that may be cut. That would mean creating an alternative shell, which is way more advanced on Windows than it is on Linux because the Windows kernel is all closed source... as is the Windows NT ui. I might be wrong about some of this, but it looks bleak in regard to things that MS is forcefully cutting out of the OS. They are making a huge mistake with Windows 11 in forcing everyone to use an interface that is really for tablet PCs.

**Comment by DavidXanatos** [comment]
its not that difficult see
https://github.com/cairoshell/cairoshell
and 
https://github.com/Pdawg-bytes/GyroShell
booth shell replacements for windows...

**Reply by wardhale** [reply]
I didn't say it was impossible... but it's way more difficult than it is on Linux. I'd imagine that's why they have like 100 guis and Windows has a only a small few, mostly unmaintained these days. Windows does have stuff like Windowblinds, but that doesn't change functionality too much... and it's paid.

**Reply by pyrates999** [reply]
I prefer cygwin64.  It's a bash shell for windows.
